SharedCryptoPkg/Include/Private/Shared/SharedCrypto.h

```diff
@@ -0,0 +1,1203 @@
+
```

Comment:
I see a single structure is used for the PPI, DXE Protoco, and SMM Protocol.  Instead of a shared folder, move the shared content to the Protocol directory and use a type name for the DXE Protocol.  Then from the PPI and SMM Protocol include files, include the Protocol and typedef the PPI and SMM Protocol from the DXE Protocol.

I like the use of Private to not make the PPI, Protocol, and SMM Protocol public.

Reply:
Alright- I restructured things a little bit that hopefully lines up with what you mentioned in your comment.